`src/Makefile.am` is missing `LIBADD` for dependent libraries

`configure.ac` checks for and appends required libraries to the `@X_LIBS@` variable, and then uses that in `examples/Makefile.am`'s `LDADD`, which is great for compiling the examples in-project.

But the generated `libgle.so.3.1.0` does not link with `@X_LIBS@` and so `ldd src/.libs/libgle.so.3.1.0` is missing dependency information on libGL, libGLU, libglut, etc.

This manifests as link-time errors:

`gcc -o test test.c -lgle`:
```
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/13/../../../x86_64-linux-gnu/libgle.so: undefined reference to `gluNewTess'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/13/../../../x86_64-linux-gnu/libgle.so: undefined reference to `gluTessVertex'
```

(actual example code:  https://salsa.debian.org/debian/gle/-/blob/debian/debian/tests/build   and build log: https://salsa.debian.org/debian/gle/-/jobs/4572587 )

Putting all of `@X_LIBS@` into `src/Makefile.am`'s `libgle_la_LIBADD` is almost correct, except that configure.ac also inserts `-lgle` in `X_LIBS`.   https://github.com/linas/glextrusion/blob/fb75bfce1632cdde2c17ab5083384f1dcabb8ba2/configure.ac#L141

I think a fix could be to stop inserting `-lgle` into `X_LIBS`, using `X_LIBS` to set `libgle_la_LIBADD` in `src/Makefile.am`, and inserting -lgle in only the examples/Makefile.am LIBADD variable.

--- a/configure.ac
+++ b/configure.ac
@@ -138,7 +138,7 @@ AC_CHECK_LIB(glut, glutSwapBuffers,
 #       cc @X_LIBS@ (-lfoo...) @X_PRE_LIBS@ -lX11 @X_EXTRA_LIBS@
 # (Perhaps X_LIBS should have been called X_LDFLAGS.)
 
-X_LIBS="-lgle $X_LIBS -lXt -lX11"
+X_LIBS="$X_LIBS -lXt -lX11"
 
 AC_OUTPUT(
           Makefile

--- a/examples/Makefile.am
+++ b/examples/Makefile.am
@@ -56,7 +56,7 @@ AM_CPPFLAGS = \
   -I${top_srcdir} \
   -I${top_srcdir}/src 
 
-LDADD = @X_LIBS@  -L../src/
+LDADD = -lgle @X_LIBS@  -L../src/
 
 docdir = $(datadir)/doc/gle/examples
 

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -22,6 +22,7 @@ libgle_la_SOURCES = 	\
 
 
 libgle_la_LDFLAGS = -version-info 4:0:1
+libgle_la_LIBADD = @X_LIBS@
 
 # Make sure we install gle.h into /usr/include/GL/gle.h or thereabouts
 pkgincludedir = $(includedir)/GL

